scripts/generate_api.py

```diff
@@ -287,3 +290,34 @@ def return_value(value: Any) -> List[str]:
 
 api_globals = globals()
 assert Serializable
+
+# Python 3.11+ does not treat default args with None as Optional anymore, this wrapper will still wrap them.
+# https://github.com/python/cpython/issues/90353
+def get_type_hints(func: Any, globalns: Any) -> Dict[str, Any]:
+    original_value = typing_get_type_hints(func, globalns)
+    if sys.version_info < (3, 11):
+        return original_value
+    for key, value in _get_defaults(func).items():
+        if original_value[key] is not Optional:
+            original_value[key] = Optional[original_value[key]]
```

Comment:
Could this overly apply optional? I think there should be a check for `value`, too, before wrapping in optional? It generates no diff, so looks fine as is for our use case, but seems like this is no longer equivalent to the built-in typing_get_type_hints < 3.11

Reply:
you are right, added the additional if check as well.